src/dapp/components/farm/Farm.tsx

```diff
@@ -227,15 +237,84 @@ export const Farm: React.FC = () => {
     );
   };
 
+  const getFarm = async (id: string) => {
+    return machineState.context.blockChain.getExploringFarm(id);
+  };
   const save = async () => {
     send("SAVE", { action: "SYNC" });
   };
 
   const safeBalance = balance.toNumber();
 
-  return (
+  const getExploringFarm = async (id: string) => {
+    send("SEARCH");
+
+    const data = await getFarm(id).catch(() => {
```

Comment:
XState uses an event driven approach which is fire and forget. It's good to decouple business logic (from the React component) and place it into the machine.

XState has an `invoke` field which allows you to run some logic when the state is entered. In this case you could move this async call inside of the `displayLoading` state.

Reply:
interesting, so i can get the data returning in the src of invoke? 
`invoke: {
        id: "",
        src: async () => { return DATA HERE }
        },
        onDone: {
          target: "exploring",
        },
        onError: {
          target: "failure",
        }`

and get it like: 
`const data = send('SEARCH')`

or i have to create a state in blockchain.ts?

Reply:
@GianfrancoCorrea saw your [comment ](https://discord.com/channels/880987707214544966/905567246452129873/929373689710006333) on discord. i havent used xstate before but for reference you might want to check `Gold.tsx`. it sends `MINE` event with payload. then in `machine.ts` > `mining` state, invoke `src: (context, event) => console.log(event)`

So for your case

Farm.tsx
```
send({ type: "SEARCH", id: id });
```

then in `invoke`
machine.ts
```
src: async ({ blockChain }, event) => console.log(event.id)
```

let me know if that helps, still in the process of learning :P

Reply:
@j923sneaks ohh dude thanks you, the last line is what i need
`(event as SearchFarmEvent).id`
I was doing it without the .id 😅